astropy/utils/diff.py

```diff
@@ -78,18 +82,31 @@ def report_diff_values(a, b, fileobj=sys.stdout, indent_width=0):
         diff_indices = np.transpose(np.where(a != b))
```

Comment:
You can maybe simplify the code using `where_not_allclose` (in the same file) since it would return directly the indices for which the difference doesn't respect the tolerance ?

Reply:
`where_not_allclose` only manages numerical values.
`test_different_table_data` in `astropy/io/fits/tests/test_diff.py`, for instance, checks the differences between boolean values and throws en error ... am I missing something?

Reply:
I didn't think about that ;). Since atol/rtol make sense only for floating array, maybe we can use `where_not_allclose` only in that case and keep `np.where(a != b)` for the other cases ?

Reply:
mmmhh... I have played a bit with `where_not_close`, but I couldn't really simplify the code.... my concern here is that arrays can have a numerical type, in that case  I can use `where_not_close` and all the elements are numerical. On the other hand arrays can have object or string type and contain float and in that case `where_not_close` will not work anyway and I should go back to the element wise check with `diff_values`.
Does that make sense?

Reply:
Yes, I was thinkg something like this
```
if (np.issubdtype(a.dtype, np.floating) and
    np.issubdtype(b.dtype, np.floating)):
    diff_indices = np.transpose(where_not_close(a, b, rtol=...))
else:
    diff_indices = np.transpose(np.where(a != b))
```
With this the changes below to show only the first 3 differences should be unnecessary ?

Reply:
Thanks! Starting using functions I'm not familiar with... sorry...